cro/cro.py

```diff
@@ -42,15 +44,38 @@ def reefinitialization (self):
         Output:
             - REEF: reef matrix
             - REEFpob: population matrix
-        """    
-        np.random.seed(seed = self.seed)
-        O = int(np.round(self.N*self.M*self.r0)) # number of occupied reefs    
-        A = np.random.randint(2, size=[O, self.L])
-        B = np.zeros([( (self.N*self.M)-O), self.L], int)
-        REEFpob = np.concatenate([A,B]) # Population creation
-        REEF = np.array((REEFpob.any(axis=1)),int) 
-        return (REEF, REEFpob)
+        """  
+        
+        # print error. Maybe use other place for all arg-checks
+        if ( (self.param_grid=={}) & (self.mode =='disc') ):
```

Comment:
It's not necessary to check whether `param_grid` equals an empty dict:
```python
if self.mode == 'disc' and not self.param_grid:
```
Maybe somthing like this is more readable?